lib/internal/url.js

```diff
@@ -647,6 +647,35 @@ function getObjectFromParams(array) {
   return obj;
 }
 
+// Mainly to mitigate func-name-matching ESLint rule
+function defineIDLClass(proto, classStr, obj) {
+  // https://heycam.github.io/webidl/#dfn-class-string
+  Object.defineProperty(proto, Symbol.toStringTag, {
+    writable: false,
+    enumerable: false,
+    configurable: true,
+    value: classStr
+  });
+
+  // https://heycam.github.io/webidl/#es-operations
+  for (const key of Object.keys(obj)) {
+    Object.defineProperty(proto, key, {
+      writable: true,
+      enumerable: true,
+      configurable: true,
+      value: obj[key]
+    });
+  }
+  for (const key of Object.getOwnPropertySymbols(obj)) {
```

Comment:
Would be easier to read if s/key/symbol/, or just separate the two mixins(take `operations` and `symbols` as arguments instead of one `obj`)